rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp

```diff
@@ -130,7 +130,12 @@ class RobotUpdateHandle
   class ActionExecution
   {
   public:
-    /// Update the amount of time remaining for this action
+    /// Update navigation or docking replan requests and remaining time estimate.
+    void update_request(
```

Comment:
I agree we don't need an `update_position()` given the `GetStateCallback` that is registered for the robot. But we also do not need this `update_request()` function. Users can call the `update_remaining_time()` that already exists below. 
To request for a replan, we can have a separate `bool replan();` function.
We could also add a function to enable/disable automatic interpolation of remaining duration which was the point of having the `std::optional<rmf_traffic::Duration>` iirc

Reply:
Added a `void replan(bool request_replan)` in [8765d80](https://github.com/open-rmf/rmf_ros2/pull/235/commits/8765d80367be1d29ab883f1cd863a615712cd214). 
About the function to enable/disable remaining duration calculation, do you mean something like what is implemented [here](https://github.com/open-rmf/rmf_ros2/blob/feature/easy_full_control/rmf_fleet_adapter/src/rmf_fleet_adapter/agv/EasyFullControl.cpp#L757-L771)?